diff --git a/src/gallium/drivers/nouveau/codegen/unordered_set.h b/src/gallium/drivers/nouveau/codegen/unordered_set.h
index 65bbec6..d95aff7 100644
--- a/src/gallium/drivers/nouveau/codegen/unordered_set.h
+++ b/src/gallium/drivers/nouveau/codegen/unordered_set.h
@@ -1,7 +1,7 @@
 #ifndef __NV50_UNORDERED_SET_H__
 #define __NV50_UNORDERED_SET_H__
 
-#if (__cplusplus >= 201103L) || defined(ANDROID)
+#if (__cplusplus >= 201103L) || defined(__clang__) || defined(ANDROID)
 #include <unordered_set>
 #else
 #include <tr1/unordered_set>
@@ -9,7 +9,7 @@
 
 namespace nv50_ir {
 
-#if __cplusplus >= 201103L
+#if __cplusplus >= 201103L || defined(__clang__)
 using std::unordered_set;
 #elif !defined(ANDROID)
 using std::tr1::unordered_set;

diff --git a/src/gallium/drivers/nouveau/codegen/nv50_ir_ra.cpp b/src/gallium/drivers/nouveau/codegen/nv50_ir_ra.cpp
index d36c853..8f24454 100644
--- a/src/gallium/drivers/nouveau/codegen/nv50_ir_ra.cpp
+++ b/src/gallium/drivers/nouveau/codegen/nv50_ir_ra.cpp
@@ -26,7 +26,7 @@
 #include <algorithm>
 #include <stack>
 #include <limits>
-#if __cplusplus >= 201103L
+#if __cplusplus >= 201103L || defined(__clang__)
 #include <unordered_map>
 #else
 #include <tr1/unordered_map>
@@ -34,7 +34,7 @@
 
 namespace nv50_ir {
 
-#if __cplusplus >= 201103L
+#if __cplusplus >= 201103L || defined(__clang__)
 using std::hash;
 using std::unordered_map;
 #elif !defined(ANDROID)
